### PR TITLE
[AWS Fargate] Add memory hard limit from container metadata and remove usage percentage

### LINKED ISSUE #37120: [AWS Fargate] Add Memory Hard Limit from container metadata
Currently, not all metadata from the containers is being parsed by the metricbeat awsfargate module:

https://github.com/elastic/beats/blob/17d5bb1ff973322269ae428c138359baefc79bdc/x-pack/metricbeat/module/awsfargate/task_stats/container.go#L12-L18

The following line should be added to extract the Hard Limit that will be used in the integration dashboard:

> `Limits  map[string]string `

### PR DESCRIPTION
**Pull request by lucian-ioan**
## Proposed commit message
Add memory hard limit from container metadata and remove usage percentage.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## How to test this PR locally

1. Follow the AWS Fargate [blogpost](https://www.elastic.co/blog/elastic-agent-monitor-ecs-aws-fargate-elastic-observability) to create to set up the integration and create the resources.

2. Be sure that your resource has cmd execution rights, for example add `"enable-execute-command"` argument while creating the resource via the AWS CLI:

```
aws ecs create-service \
    --cluster cluster-name \
    --task-definition task-definition-name \
    --enable-execute-command \
    --service-name service-name \
    --desired-count 1
```

3.  Run the shell via AWS CLI via:
```
aws ecs execute-command --cluster cluster-name \
    --task task-id \
    --container container-name \
    --interactive \
    --command "/bin/sh"
```

4. Retrieve ECS_CONTAINER_METADATA_URI_V4 via ```echo $ECS_CONTAINER_METADATA_URI_V4```.

5. Fetch task data via `curl`:
```
curl $ECS_CONTAINER_METADATA_URI_V4/task
curl $ECS_CONTAINER_METADATA_URI_V4/task/stats
```

6. Check data with the one fetched from the agent deployed in AWS (note that the agent needs to be built from the code in this PR).

## Related issues

- Closes #37120

### CODE REVIEW

**Comment by lalit-satapathy** [comment]
CC: @agithomas @shub1503 to planing the review once the PR is opened.

**Comment by lucian-ioan** [comment]
@agithomas, @shmsr, this is ready for review.

File: x-pack/metricbeat/module/awsfargate/task_stats/container.go [hunk 1]
```diff
@@ -15,6 +15,7 @@ type container struct {
 	Name     string
 	Image    string
 	Labels   map[string]string
+	Limits   Limits `json:"Limits"`
```
Comment (shmsr) [inline x-pack/metricbeat/module/awsfargate/task_stats/container.go:18]: nit: As the JSON key is the same as the struct field name, we can remove the tag `json:"Limits"` like other field names in the same struct

**Review by shmsr — APPROVED** [review]
LGTM. Suggested a couple of nits.

**Review by agithomas — COMMENTED** [review]
Can you highlight it is the Hard Limit , to avoid confusion?

File: x-pack/metricbeat/module/awsfargate/task_stats/data.go [hunk 2]
```diff
@@ -57,6 +57,11 @@ func createEvent(stats *Stats) mb.Event {
 		_, _ = e.MetricSetFields.Put("task_known_status", taskKnownStatus)
 	}
 
+	memoryHardLimit := stats.Container.Limits.Memory
+	if memoryHardLimit != 0 {
+		_, _ = e.MetricSetFields.Put("memory_hard_limit", taskKnownStatus)
```
Comment (shmsr) [inline x-pack/metricbeat/module/awsfargate/task_stats/data.go:62]:
```suggestion
		_, _ = e.MetricSetFields.Put("memory_hard_limit", memoryHardLimit)
```

**Comment by gpop63** [comment]
@lucian-ioan you were able to get actual container limits while testing this with a custom built agent containing these changes, right?

**Comment by lucian-ioan** [comment]
@gpop63 yes, tested multiple times and the memory limits line up with what I set in AWS.

**Review by gpop63 — APPROVED** [review]

**Pull request merged**